### PR TITLE
Use IResourceStore for localisation

### PR DESCRIPTION
**Pull request by huoyaoyuan**
Seems methods in `LocalisationEngine` can be non-virtual, as variant works are done in the storages class.
Prefer `zh-Hans` to `zh-CHS` because the latter is called "legacy".
How do we implement "download after selection"?

### CODE REVIEW

**Comment by kj415j45** [comment]
about zh-HanS, zh-CHS, zh-CN, ............
see also:https://msdn.microsoft.com/en-us/library/system.globalization.cultureinfo.createspecificculture.aspx

**Comment by huoyaoyuan** [comment]
@kj415j45 
`zh-CN`: Chinese (Simplified, China)
`zh-CHS`: Chinese (Simplified) Legacy
`zh-Hans`: Chinese (Simplified)

For compatibility, there's a `zh-CN`->`zh-CHS`->`zh-Hans` deriving order.
Current localization text of _osu!_ is using `zh-CHS`, so I suggested to change.

File: osu.Framework/Localisation/FormattableString.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    /// A bindable string constructed from <see cref="string.Format(string, object[])"/>.
+    /// </summary>
+    public class FormattableString : Bindable<string>
+    {
+        private readonly string format;
+        private readonly object[] objects;
+        protected virtual string Format => format;
+        public void Update() => Value = string.Format(Format, objects);
+
+        public FormattableString(string format, object[] objects)
```
Comment (peppy) [inline osu.Framework/Localisation/FormattableString.cs]: i think you may wanna use `params` here

**Comment by peppy** [comment]
I'm not sure we need to do download-after-selection. I'm hoping we can just distribute the game with all languages bundled this time around (osu-stable is only 3.5mb *before* compression to include all locales).

**Comment by peppy** [comment]
nice work 👍

**Pull request merged**